recipe-client-addon/test/unit/test_NormandyApi.js

```diff
@@ -126,6 +142,73 @@ add_task(withMockApiServer(async function test_fetchRecipes() {
   equal(recipes[0].name, "system-addon-test");
 }));
 
+add_task(async function test_fetchRecipes_canonical_mismatch() {
+  const getApiUrl = sinon.stub(NormandyApi, "getApiUrl").resolves("http://localhost/recipes/");
+
+  // Recipe is non-canonical (it has whitespace, properties are out of order)
+  const response = new MockResponse(`[
+    {
+      "recipe": {"b": 1, "a": 2},
+      "signature": {"signature": "", "x5u": ""}
+    }
+  ]`);
+  const get = sinon.stub(NormandyApi, "get").resolves(response);
+
+  try {
+    await NormandyApi.fetchRecipes();
+    ok(false, "fetchRecipes did not throw for canonical JSON mismatch");
+  } catch (err) {
+    ok(err instanceof InvalidSignatureError, "Error was not an InvalidSignatureError");
```

Comment:
I don't have much of a preference except that we should document a preference. Right now we're inconsistent across the codebase. I'll roll with your suggestion.

(Well, that's a lie; my preference is the style we use with pytest, where assertions generally use the auto-generated "blah != foo" message unless the assertion is tricky enough to need more context, but that relies heavily on the context that comes with pytest's output.)